File: 2022.01.21/javascript/2798.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const [n,m] = input[0].split(' ').map((a) => {return Number(a)});
const nums = input[1].split(' ').map(Number);
let num, result = 0;
for (let i = 0 ; i<nums.length-2 ; i++){
    for (let j = i+1 ; j<nums.length-1 ; j++){
        for (let k = j+1 ; k < nums.length ; k++){
            num = nums[i]+nums[j]+nums[k];
            if (m>=num && num>result) result = num;
            if (result == m) break;
        }
        if (result == m) break;
    }
    if (result == m) break;
}
console.log(result)

File: Boj.단계별/11_브루트포스/python/2798.py
n,m = map(int,input().split())
nums = list(map(int,input().split()))
result = 0
for i in range(len(nums)-2):
    for j in range(i+1,len(nums)-1):
        for k in range(j+1,len(nums)):
            temp = nums[i]+nums[j]+nums[k]
            if m>=temp>result:
                result = temp
print(result)